domain/storageprovisioning/service/volume.go

```diff
@@ -110,14 +165,212 @@ type VolumeState interface {
+	defer span.End()
+
+	if err := uuid.Validate(); err != nil {
+		return 0, errors.Errorf(
+			"validating volume attachment uuid: %w", err,
+		).Add(coreerrors.NotValid)
+	}
+
+	life, err := s.st.GetVolumeAttachmentLife(ctx, uuid)
+	if err != nil {
+		return 0, errors.Capture(err)
+	}
+	return life, nil
+}
+
+// GetVolumeAttachmentUUIDForVolumeIDMachine returns the volume attachment
+// uuid for the supplied volume ID which is attached to the machine.
+//
+// The following errors may be returned:
+// - [corestorage.InvalidStorageID] when the provided id is not valid.
+// - [coreerrors.NotValid] when the provided machine uuid is not valid.
+// - [storageprovisioningerrors.VolumeNotFound] when no volume exists for the
+// supplied id.
+// - [storageprovisioningerrors.VolumeAttachmentNotFound] when no volume
+// attachment exists for the supplied values.
+// - [machineerrors.MachineNotFound] when no machine exists for the provided
+// machine uuid.
+func (s *Service) GetVolumeAttachmentUUIDForVolumeIDMachine(
+	ctx context.Context,
+	id corestorage.ID,
+	machineUUID coremachine.UUID,
+) (storageprovisioning.VolumeAttachmentUUID, error) {
+	ctx, span := trace.Start(ctx, trace.NameFromFunc())
+	defer span.End()
+
+	if err := id.Validate(); err != nil {
+		return "", errors.Capture(err)
+	}
+	if err := machineUUID.Validate(); err != nil {
+		return "", errors.Capture(err)
+	}
+
+	netNodeUUID, err := s.st.GetMachineNetNodeUUID(ctx, machineUUID)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	vUUID, err := s.st.GetVolumeUUIDForID(ctx, id.String())
+	if err != nil {
+		return "", errors.Errorf(
+			"getting volume uuid for id %q: %w", id.String(), err,
+		)
+	}
+
+	uuid, err := s.st.GetVolumeAttachmentUUIDForVolumeNetNode(
+		ctx, vUUID, netNodeUUID,
+	)
+	if errors.Is(err, networkerrors.NetNodeNotFound) {
+		return "", errors.Errorf(
+			"machine %q does not exist", machineUUID.String(),
+		).Add(machineerrors.MachineNotFound)
+	} else if errors.Is(err, storageprovisioningerrors.VolumeNotFound) {
+		return "", errors.Errorf(
+			"volume %q does not exist", id.String(),
+		).Add(storageprovisioningerrors.VolumeNotFound)
+	} else if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	return uuid, nil
+}
+
+// GetVolumeAttachmentUUIDForVolumeUnit returns the volume attachment uuid
+// for the supplied volume ID which is attached to the unit.
+//
+// The following errors may be returned:
+// - [corestorage.InvalidStorageID] when the provided id is not valid.
+// - [coreerrors.NotValid] when the provided unit uuid is not valid.
+// - [storageprovisioningerrors.VolumeNotFound] when no volume exists for the
+// supplied id.
+// - [storageprovisioningerrors.VolumeAttachmentNotFound] when no volume
+// attachment exists for the supplied values.
+// - [applicationerrors.UnitNotFound] when no unit exists for the provided unit
+// uuid.
+func (s *Service) GetVolumeAttachmentUUIDForVolumeUnit(
+	ctx context.Context,
+	id corestorage.ID,
```

Comment:
In the facade, what we have is volumeTag(`volume-666`, for example), which only has the serial number.
So here we should use volumeID string instead of `corestorage.ID`(format: "name/int").